packages/openfdd-agent-shell/src/openfdd_agent_shell/cron/scheduler.py

```
@@ -196,6 +197,46 @@ def _execute(self, job: CronJob) -> str:
+                user = str(payload.get("user") or os.environ.get("OFDD_AGENT_USER") or "agent")
+                password = str(payload.get("password") or os.environ.get("OFDD_AGENT_PASSWORD") or "")
+                if not password:
+                    raise ValueError("health_building_agent requires token or agent password")
+                login_body = json.dumps({"username": user, "password": password}).encode("utf-8")
+                req = urllib.request.Request(
+                    f"{base}/api/auth/login",
+                    data=login_body,
+                    headers={"Content-Type": "application/json"},
+                    method="POST",
+                )
+                with urllib.request.urlopen(req, timeout=30) as resp:
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)